internal/test/helpers/vmware/intg_test_context.go

```diff
@@ -138,34 +139,41 @@ func NewIntegrationTestContextWithClusters(ctx context.Context, integrationTestC
 						APIVersion: clusterv1.GroupVersion.String(),
 						Kind:       "Cluster",
 						Name:       testCtx.Cluster.Name,
-						UID:        testCtx.Cluster.UID,
+						// Using a random id in this case is okay because we don't rely on it to be the correct uid.
+						UID: "should-be-uid-of-cluster",
 					},
 				},
 			},
 			Data: map[string][]byte{
 				"value": buf,
 			},
 		}
-		Expect(integrationTestClient.Create(ctx, secret)).To(Succeed())
-		Eventually(func() error {
-			return integrationTestClient.Get(ctx, client.ObjectKeyFromObject(secret), secret)
-		}).Should(Succeed())
+		testCtx.KubeconfigSecret = secret
 	})
 
 	By("Create a vsphere cluster and wait for it to exist", func() {
```

Comment:
By seems misleading. Should we change this here to generate? (same in all other cases around here where we dont call Create)

(also the func names please, e.g. createCluster)